fem/coefficient.cpp

```diff
@@ -775,4 +775,93 @@ double ComputeGlobalLpNorm(double p, VectorCoefficient &coeff, ParMesh &pmesh,
 }
 #endif
 
+VectorQuadratureFunctionCoefficient::VectorQuadratureFunctionCoefficient(
+   QuadratureFunction *qf)
+   : VectorCoefficient(qf->GetVDim()), QuadF(qf), index(0),
+     length(qf->GetVDim()) {}
+
+void VectorQuadratureFunctionCoefficient::SetQuadratureFunction(
+   QuadratureFunction *qf)
+{
+   MFEM_VERIFY(QuadF, "QuadratureFunction must be set to a nonnull ptr");
+
+   index = 0;
+   length = qf->GetVDim();
+   vdim = length;
+   QuadF = qf;
+}
+
+void VectorQuadratureFunctionCoefficient::SetComponent(int _index, int _length)
+{
+   MFEM_VERIFY(_index >= 0, "Index must be >= 0");
+   MFEM_VERIFY(_index < QuadF->GetVDim(),
+               "Index must be < QuadratureFunction length");
+   index = _index;
+
+   MFEM_VERIFY(_length > 0, "Length must be > 0");
+
+   int diff = QuadF->GetVDim() - index;
+   MFEM_VERIFY(_length <= diff,
+               "Length must be <= (QuadratureFunction length - index)");
+
+   length = _length;
+   vdim = length;
+}
+
+void VectorQuadratureFunctionCoefficient::Eval(Vector &V,
+                                               ElementTransformation &T,
+                                               const IntegrationPoint &ip)
+{
+   MFEM_VERIFY(QuadF, "QuadratureFunction must be set to a nonnull ptr");
+
+   QuadF->HostRead();
+   int elem_no = T.ElementNo;
+   if (index == 0 && length == QuadF->GetVDim())
+   {
+      Vector temp;
+      QuadF->GetElementValues(elem_no, ip.index, temp);
+      V = temp;
+   }
+   else
+   {
+      Vector temp;
+      QuadF->GetElementValues(elem_no, ip.index, temp);
+      double *data = temp.HostReadWrite();
```

Comment:
Why is `Write` needed here? Actually should there be any call at all, as there's one on line 817?

Reply:
It wasn't needed. The code was restructured so this was no longer needed.